src/gateway/http.js

```diff
@@ -69,8 +69,34 @@ HTTP.prototype = Gateway.extends({
       assign(requestParams, httpOptions.configure(requestParams))
     }
 
+    if (httpOptions.onRequestWillStart) {
```

Comment:
Was not sure about this callback. The client could technically use the `configure` callback instead but that's maybe not as clear?